### ISSUE TITLE
flask web is 404 error

### ISSUE THREAD

**Issue by winston11tw** [sean-mcclure/defect_classifier#1]
When I input python api/app.py
the code run successfully 
but the web is error
![404](https://user-images.githubusercontent.com/62945333/78033341-d987cb80-7398-11ea-8fbc-d5d8a7c7c730.png)
![error](https://user-images.githubusercontent.com/62945333/78033354-de4c7f80-7398-11ea-8846-3fe54e72f10a.png)
How can I solve this problem?

**Comment by sean-mcclure** [comment]
Apologies, the Medium article was missing the instruction to **start a web server.** I have updated the Medium article [here](https://bit.ly/2QY5EEj).  

So you **don't open the URL from Flask**, you open the GUI contained in **app/index.html.** 

Step 1: Start a web service by running in terminal:

`python3 -m http.server`

Step 2: Point your browser to:

`http://localhost:8000/app/`

The GUI will connect as needed.

**Comment by winston11tw** [comment]
Thanks. According to Medium article, now can open the web and FETCH BOARDS.
But when I press RUN PIPELINE, the defect classfier will continue loading and result is not display.
like this:
![error2](https://user-images.githubusercontent.com/62945333/78052914-5f644080-73b2-11ea-8f67-4a619f1bfc4f.PNG)
I have already move the export.pkl to /model 
Are there any step that  I still missing ?
Thanks for your help~

**Comment by kedionai** [comment]
Can you inspect your browser console (inspect element) to see if there are any errors?

**Comment by winston11tw** [comment]
@kedionai  
![error3](https://user-images.githubusercontent.com/62945333/78061352-b3295680-73bf-11ea-810e-e2a12d93594a.PNG)
the CORS error

**Comment by sean-mcclure** [comment]
Hmm that's odd. I do not have the CORS issue, and it shouldn't happen with localhost since back-end and front-end are on same server. **Which browser are you using**? I know some people have had [issues](https://stackoverflow.com/questions/10883211/deadly-cors-when-http-localhost-is-the-origin) with Chrome browser and localhost when it comes to CORs, however I am using Chrome (Mac) with no issues.

**Comment by winston11tw** [comment]
I was using Firefox web browser which was default in Ubuntu.
And the "python3 -m http.server" is run on the anaconda environment.
Maybe I should  install Chrome and try again.
![error4](https://user-images.githubusercontent.com/62945333/78065791-5da47800-73c6-11ea-92d0-eb27b05351c5.PNG)
Chrome still has the problem.

**Comment by sean-mcclure** [comment]
I think it's something with your environment. You might have to look into conda and/or Ubuntu issues with CORs. Since you are using Ubuntu, are you sure your browser is on the same machine as the server?

**Comment by winston11tw** [comment]
I check the web console again, and find the call_function's local host is 5000.
![8002](https://user-images.githubusercontent.com/62945333/78111790-c96efb00-742f-11ea-953a-fe4e69b2d8a0.PNG)
Other function's local host is 8000. 
![7999](https://user-images.githubusercontent.com/62945333/78111808-d2f86300-742f-11ea-86b2-572599251401.PNG)
So I change the code in call_api.js, from "url": "http://localhost:5000/call_function/" to http://localhost:8000/call_function/ and try again.
Now the CORS error is solved, but become the 404 error.
![8000OK](https://user-images.githubusercontent.com/62945333/78113067-c07f2900-7431-11ea-9ebf-93f2f2671391.PNG)

**Comment by kedionai** [comment]
The call function is supposed to be 5000, since thats the endpoint port from Flask. It’s the CORS issue that needs to be resolved.

**Comment by kedionai** [comment]
Another option is I could add a Dockerfile, and see if you can access the application from outside a Docker container.

**Comment by winston11tw** [comment]
@kedionai  
Can you try it? thanks~

**Comment by sean-mcclure** [comment]
The fastai doesnt seem to install properly using Docker. I'll let you know if I figure it out.

**Comment by trucnguyen83** [comment]
![image](https://user-images.githubusercontent.com/56549047/110417429-936ed800-80d0-11eb-81ce-be7dcb43f122.png)
having these errors when clicking run pipeline. Have anyone solved the cors issue?

**Comment by trucnguyen83** [comment]
Hi,
I clone the latest version but got the GET http://localhost:8000/app/ 404 (File not found) error when viewing in browser. 

I saw your config.js is
az.hold_value.config = {
    api_url : "https://5000-violet-goose-u4jo2x0n.ws-us03.gitpod.io"}

Do we need to change this url to "http://localhost:5000"?  I tried but still cannot work

**Comment by kedionai** [comment]
Yes you must change the URL in config.js to your own localhost and port; usually localhost:8000

Are you running api service from inside the api directory?

Sean McClure, Ph.D.
Founder/CEO, Kedion Inc.
kedion.ai
415-745-3688
